mlos_bench/mlos_bench/services/local/temp_dir_context.py

```diff
@@ -62,6 +70,8 @@ def temp_dir_context(self, path: Optional[str] = None) -> Union[TemporaryDirecto
         temp_dir_context : TemporaryDirectory
             Temporary directory context to use in the `with` clause.
         """
-        if path is None and self._temp_dir is None:
+        temp_dir = path or self._temp_dir
+        if temp_dir is None:
             return TemporaryDirectory()
-        return nullcontext(path or self._temp_dir)
+        os.makedirs(temp_dir, exist_ok=True)
```

Comment:
I don't like that we create a new directory right inside the context - and never delete it. The original behavior was to either use some existing, externally provided path (and leave it to the client to clean up) or create a new (temp) directory and take the responsibility for cleaning it. OTOH, I understand that this is probably the most convenient place to create a temp_dir, and I am not sure if the class should clean it on the context's `__exit__()` - that can easily screw things up, and it will be very hard to debug. So maybe let's keep `os.makedirs()` here. End of rant. 😄

Reply:
My thought was that if someone specifies in their config a known location, they don't want it to be cleaned up.
It's kind of known not to be a "temp" directory then.
So, in that case, rather than error out when that known location doesn't already exist, we can "just" create it for them, but leave the rest to them to manage.